spark/v3.4/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestViews.java

```diff
@@ -2109,6 +2109,34 @@ public void createViewWithCustomMetadataLocation() throws IOException {
                 ""));
   }
 
+  @Test
+  public void readFromViewVersionTable() throws NoSuchTableException {
+    insertRows(10);
+    String viewName = viewName("simpleView");
+    String sql = String.format("SELECT id FROM %s", tableName);
+
+    ViewCatalog viewCatalog = viewCatalog();
+
+    viewCatalog
+        .buildView(TableIdentifier.of(NAMESPACE, viewName))
+        .withQuery("spark", sql)
+        // use non-existing column name to make sure only the SQL definition for spark is loaded
+        .withQuery("trino", String.format("SELECT non_existing FROM %s", tableName))
+        .withDefaultNamespace(NAMESPACE)
+        .withDefaultCatalog(catalogName)
+        .withSchema(schema(sql))
+        .create();
+
+    // Similar to table's metadata table, view's metadata table requires fully qualified name.
+    List<Object[]> result = sql("SELECT * FROM %s.%s.%s.version", catalogName, NAMESPACE, viewName);
+    assertThat(result).hasSize(1);
+    assertThat(result.get(0).length).isEqualTo(7);
```

Comment:
we should add some additional checks here to assert all the values are what we expect. For this, you can store the result of the view creation above in a `view` variable and then use `view.currentVersion()` to compare values

Reply:
Done